ext/js/data/database.js

```diff
@@ -43,7 +43,9 @@ export class Database {
         try {
             this._isOpening = true;
             this._db = await this._open(databaseName, version, (db, transaction, oldVersion) => {
-                this._upgrade(db, transaction, oldVersion, structure);
+                if (structure !== null) {
```

Comment:
This should probably be part of the main PR

Reply:
No, the PR makes this field nullable so that we can have workers that interact with the DB but don't upgrade it. It's a logic change.